weather-booking-app-client/src/pages/BookingPage/BookingPage.tsx

```diff
@@ -139,12 +140,13 @@ class BookingPage extends Component<BookingPageProps, BookingPageState> {
         wind: this.state.selectedWindOption,
         temperature: this.state.selectedTemperatureOption,
       }
-    );
-    setTimeout(() => {}, 500);
-
-    this.redirectToBookListPage();
+    ).then(() => {
+      setTimeout(() => {
+        this.redirectToBookListPage();
+      }, 1000);
```

Comment:
Why do we need a second delay?

Reply:
because otherwise it redirects to ViewBookingsPage too quickly and then it fetches the data before the POST booking api request has completed. So the most recent booking won't appear. Its definitely scuffed. If you know better way to handle this then your welcome to change it